activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -2,6 +2,16 @@
 require 'active_support/core_ext/object/acts_like'
 
 module ActiveSupport
+  class NoMethodOnObjectError < NoMethodError
+    attr_reader :object
+
+    def initialize(msg, name, args = nil, object = nil)
+      super(msg, name, args)
+
+      @object = object
+    end
+  end
+
```

Comment:
What about a file just for this class, and some documentation too? IMO it's sufficiently generic to other people use if needed.